src/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -0,0 +1,395 @@
+			if (isTownyAdmin) {
+				townSpawnPermission = TownSpawnLevel.ADMIN;
+			} else if ((split.length == 0) && (!outpost)) {
+				townSpawnPermission = TownSpawnLevel.TOWN_RESIDENT;
+			} else {
+				// split.length > 1
+				if (!resident.hasTown()) {
+					townSpawnPermission = TownSpawnLevel.UNAFFILIATED;
+				} else if (resident.getTown() == town) {
+					townSpawnPermission = outpost ? TownSpawnLevel.TOWN_RESIDENT_OUTPOST : TownSpawnLevel.TOWN_RESIDENT;
+				} else if (resident.hasNation() && town.hasNation()) {
+					Nation playerNation = resident.getTown().getNation();
+					Nation targetNation = town.getNation();
+
+					if (playerNation == targetNation) {
+						if (!town.isPublic() && TownySettings.isAllySpawningRequiringPublicStatus())
+							throw new TownyException(
+									String.format(TownySettings.getLangString("msg_err_ally_isnt_public"), town));
+						else
+							townSpawnPermission = TownSpawnLevel.PART_OF_NATION;
+					} else if (targetNation.hasEnemy(playerNation)) {
+						// Prevent enemies from using spawn travel.
+						throw new TownyException(TownySettings.getLangString("msg_err_public_spawn_enemy"));
+					} else if (targetNation.hasAlly(playerNation)) {
+						if (!town.isPublic() && TownySettings.isAllySpawningRequiringPublicStatus())
+							throw new TownyException(
+									String.format(TownySettings.getLangString("msg_err_ally_isnt_public"), town));
+						else
+							townSpawnPermission = TownSpawnLevel.NATION_ALLY;
+					} else {
+						townSpawnPermission = TownSpawnLevel.UNAFFILIATED;
+					}
+				} else {
+					townSpawnPermission = TownSpawnLevel.UNAFFILIATED;
+				}
+			}
+
+			TownyMessaging.sendDebugMsg(townSpawnPermission.toString() + " " + townSpawnPermission.isAllowed(town));
+			townSpawnPermission.checkIfAllowed(plugin, player, town);
+
+			// Check the permissions
+			if (!(isTownyAdmin || ((townSpawnPermission == TownSpawnLevel.UNAFFILIATED) ? town.isPublic()
+					: townSpawnPermission.hasPermissionNode(plugin, player, town))))
+				throw new TownyException(TownySettings.getLangString("msg_err_not_public"));
+
+			// Prevent outlaws from spawning into towns they're considered an outlaw in.
+			if (!isTownyAdmin) {
+				if (town.hasOutlaw(resident))
+					throw new TownyException(String.format(
+							TownySettings.getLangString("msg_error_cannot_town_spawn_youre_an_outlaw_in_town"), town));
+			}
+
+			break;
+		case NATION:
+			nation = (Nation) townyObject;
+			spawnLoc = nation.getNationSpawn();
+
+			// Determine conditions
+			if (isTownyAdmin) {
+				nationSpawnPermission = NationSpawnLevel.ADMIN;
+			} else if ((split.length == 0)) {
+				nationSpawnPermission = NationSpawnLevel.PART_OF_NATION;
+			} else {
+				// split.length > 1
+				if (!resident.hasTown()) {
+					nationSpawnPermission = NationSpawnLevel.UNAFFILIATED;
+				} else if (resident.hasNation()) {
+					Nation playerNation = resident.getTown().getNation();
+					Nation targetNation = nation;

```

Comment:
`targetNation` is useless. Why not just use the `nation` variable? Or just rename `nation` to `targetNation`.